src/test/java/io/github/sashirestela/openai/SimpleOpenAITest.java

```diff
@@ -31,45 +31,45 @@ class SimpleOpenAITest {
     HttpClient httpClient = mock(HttpClient.class);
     CleverClient cleverClient = mock(CleverClient.class);
 
-    @Test
-    void shouldSetPropertiesToDefaultValuesWhenBuilderIsCalledWithoutThoseProperties() {
-        var openAI = SimpleOpenAI.builder()
-                .apiKey("apiKey")
-                .build();
-        assertEquals(HttpClient.Version.HTTP_2, openAI.getHttpClient().version());
-        assertEquals(OPENAI_BASE_URL, openAI.getBaseUrl());
-        assertNotNull(openAI.getCleverClient());
-    }
-
-    @Test
-    void shouldSetPropertiesWhenBuilderIsCalledWithThoseProperties() {
-        var otherUrl = "https://openai.com/api";
-        var openAI = SimpleOpenAI.builder()
-                .apiKey("apiKey")
-                .baseUrl(otherUrl)
-                .httpClient(httpClient)
-                .build();
-        assertEquals("apiKey", openAI.getApiKey());
-        assertEquals(otherUrl, openAI.getBaseUrl());
-        assertEquals(httpClient, openAI.getHttpClient());
-    }
-
-    @Test
-    void shouldNotAddOrganizationToHeadersWhenBuilderIsCalledWithoutOrganizationId() {
-        var openAI = SimpleOpenAI.builder()
-                .apiKey("apiKey")
-                .build();
-        assertFalse(openAI.getCleverClient().getHeaders().containsValue(openAI.getOrganizationId()));
-    }
-
-    @Test
-    void shouldAddOrganizationToHeadersWhenBuilderIsCalledWithOrganizationId() {
-        var openAI = SimpleOpenAI.builder()
-                .apiKey("apiKey")
-                .organizationId("orgId")
-                .build();
-        assertTrue(openAI.getCleverClient().getHeaders().containsValue(openAI.getOrganizationId()));
-    }
+//    @Test
```

Comment:
these tests check for set properties. they use getters for fields that don't exist anymore. it also seems a little awkward to have a getter only for tests. I think these tests don't provide much value. but, if you think it is useful we can restore them and add the getters again.

Reply:
All the existing tests pass except for these tests that check for fields that don't exist anymore. the organizationId is embedded into a header in the new design.

with your permission I'll delete the commented out tests.

I'll for the new classes I'll add tests for the prepare functions. 

I think what can really increase the testability and the confidence that simple-openai works as expected and changes are safe is to add end-to-end tests. pretty much what the demos are doing, but against a dedicated HTTP test server, not the real providers of course. Each provider will have its own mock service that conforms to its convention and can check the headers and base url passed in and the request schema. Obviously , this is a bug change so not for today. WDYT?